src/app/modules/timesheet/components/timesheet-edit/timesheet-edit.component.ts

```diff
@@ -79,6 +91,8 @@ export class TimesheetEditComponent implements OnInit {
       this.timesheetService.timesheet.flatFees = this.generateExpenses ? this.flatFeesForm.flatFees : [];
       this.updateMailtoLink();
       this.reactToSubmition(false);
+      this.setShortUrl('edit');
+      this.setShortUrl('review');
```

Comment:
You can maybe build the both URL at the same time, I don't know if set one and not the other is a need. No ?